security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/SecurityAnalysisExecutionHandler.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.distributed;
+
+import com.google.common.io.ByteSource;
+import com.powsybl.computation.*;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.xml.NetworkXml;
+import com.powsybl.security.SecurityAnalysisParameters;
+import com.powsybl.security.execution.NetworkVariant;
+import com.powsybl.security.execution.SecurityAnalysisExecutionInput;
+import com.powsybl.security.json.JsonSecurityAnalysisParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ *
+ * Base implementation for {@link ExecutionHandler}s which may execute one or multiple {@literal itools security-analysis} command(s).
+ * The exact behaviour is provided through the constructor argument.
+ * Instances are provided by factory methods of {@link SecurityAnalysisExecutionHandlers}.
+ *
+ * <p>Specified {@link Network} variant is serialized as an XIIDM file.
+ *
+ * @author Sylvain Leclerc <sylvain.leclerc at rte-france.com>
+ */
+public class SecurityAnalysisExecutionHandler<R> extends AbstractExecutionHandler<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecurityAnalysisExecutionHandler.class);
+
+    private static final String NETWORK_FILE = "network.xiidm";
+    private static final String CONTINGENCIES_FILE = "contingencies.groovy";
```

Comment:
As ContingencyProvider is a public interface, contingencies could be describe in another format